modelql-core/src/commonMain/kotlin/org/modelix/modelql/core/IStep.kt

```diff
@@ -212,7 +225,7 @@ abstract class AggregationStep<In, Out> : MonoTransformingStep<In, Out>() {
         val flow = flow {
             emit(aggregate(input))
         }
-        return if (outputIsConsumedMultipleTimes()) {
+        return if (needsCoroutineScope()) {
             val scope = context.coroutineScope ?: throw RuntimeException("Coroutine scope required for caching of $this")
             flow.shareIn(scope, SharingStarted.Lazily, 1)
```

Comment:
Is the check `flow.assertNotEmpty() // empty flow would result in deadlock` not needed here?

Reply:
I don't think the flow can be empty here, as line 226 always emits an `IStepOutput`.

Reply:
Shouldn't we assert these "i think" things? ;-)

Reply:
> I don't think the flow can be empty here, as line 226 always emits an IStepOutput.

You right just two lines above we can see, that the flow has exactly one element.